src/components/manifold-subscription-details/manifold-subscription-details.tsx

```diff
@@ -44,10 +54,50 @@ export class ManifoldSubscriptionCreate {
   }
 
   render() {
-    if (this.data?.subscription) {
-      return <h3>{this.data.subscription.plan.displayName}</h3>;
+    if (!this.data) {
```

Comment:
So weirdly, what I was observing is that when I received a `NotFound` error from the API, `this.data` would not be null, but instead would contain `{ subscription: null }`. That's why it was written this way. Does that seem like an error in the GraphQL implementation to you?

Reply:
@davidharrigan what are your thoughts? is this how the rest of our api handles not found items?

Reply:
yes that is how the other endpoints are implemented

example from resource endpoint:
```json
{
  "errors": [
    {
      "message": "An argument is invalid, please refer to the published schema",
      "path": [
        "resource"
      ],
      "extensions": {
        "type": "InvalidArgument"
      }
    }
  ],
  "data": {
    "resource": null
  }
}
```
I don't know if that is the way it should be, but it is consistent.  I'm happy to take a look, but I don't want to diverge from the other implementations.  Are you handling NotFound differently for other endpoints?

Reply:
that was an invalid argument error, but here is a not found example from plan:
```
{
  "errors": [
    {
      "message": "The requested resource is not found",
      "path": [
        "plan"
      ],
      "extensions": {
        "type": "NotFound"
      }
    }
  ],
  "data": {
    "plan": null
  }
}
```

Reply:
for everything we are checking errors before data is ever used so I don't think there's a case where it'll be null and we're trying to use it. it seems like something that should be reviewed through because that means our generated types are kind of wrong for a lot of things. We just happen to have paths that don't expose it